### PR TITLE
Enabled doxygen validation in CI pipeline.

### PR DESCRIPTION
**Pull request by LarryOsterman**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

File: CMakePresets.json [hunk 1]
```diff
@@ -302,10 +302,9 @@
         "BUILD_SAMPLES": true
       }
     },
-
     {
       "name": "linux-basic-gcc9",
-      "displayName": "Linux GCC 9",
+      "displayName": "Linux GCC 9 ",
```
Comment (ahsonkhan) [inline CMakePresets.json:307]: Revert this change.
```suggestion
      "displayName": "Linux GCC 9",
```

File: CMakePresets.json [hunk 2]
```diff
@@ -340,7 +339,42 @@
...
       "displayName": "zz-Generate Doxygen",
       "inherits": "x64-static",
-      "cacheVariables": { "BUILD_DOCUMENTATION": true }
+      "cacheVariables": {
+        "BUILD_DOCUMENTATION": true
+      }
+    },
+    {
+      "name": "linux-basic-g++",
+      "displayName": "Linux G++ ",
```
Comment (ahsonkhan) [inline CMakePresets.json:348]: consistency:
```suggestion
      "displayName": "Linux G++",
```

File: cmake-modules/AzureDoxygen.cmake [hunk 3]
```diff
@@ -17,19 +17,33 @@ function(generate_documentation PROJECT_NAME PROJECT_VERSION)
         set(DOXYGEN_OUTPUT_DIRECTORY docs)
         set(DOXYGEN_LAYOUT_FILE ${CMAKE_SOURCE_DIR}/eng/docs/api/assets/DoxygenLayout.xml)
         set(DOXYGEN_RECURSIVE YES)
+        if (MSVC)
+          set(DOXYGEN_WARN_FORMAT "$file($line) : $text")
+        endif()
+#        set(DOXYGEN_WARN_AS_ERROR FAIL_ON_WARNINGS)
```
Comment (ahsonkhan) [inline cmake-modules/AzureDoxygen.cmake:23]: uncomment?

File: cmake-modules/AzureDoxygen.cmake [hunk 4]
```diff
@@ -17,19 +17,33 @@ function(generate_documentation PROJECT_NAME PROJECT_VERSION)
...
         set(DOXYGEN_USE_MDFILE_AS_MAINPAGE ./README.md)
         # Setting the INLINE_SOURCES tag to YES will include the body of functions,
         # classes and enums directly into the documentation.
         set(DOXYGEN_INLINE_SOURCES NO)
+        set(DOXYGEN_MARKDOWN_ID_STYLE GITHUB)
         # Skip generating docs for json, test, samples, and private files.
         set(DOXYGEN_EXCLUDE_PATTERNS
             json.hpp
+            package_version.hpp
+            dll_import_export.hpp
```
Comment (ahsonkhan) [inline cmake-modules/AzureDoxygen.cmake:33]: Why do we need to skip these two?

File: sdk/attestation/azure-security-attestation/inc/azure/attestation/attestation_client.hpp [hunk 5]
```diff
@@ -129,13 +129,14 @@ namespace Azure { namespace Security { namespace Attestation {
...
      * @param options The options to customize the client behavior.
+     * @param context The context for cancelling long running operations.
      * @return The newly created client.
      */
     static AttestationClient Create(
         std::string const& endpoint,
         std::shared_ptr<Core::Credentials::TokenCredential const> credential,
         AttestationClientOptions const& options = AttestationClientOptions{},
-        Azure::Core::Context const& constext = Azure::Core::Context{});
+        Azure::Core::Context const& context = Azure::Core::Context{});
```
Comment (ahsonkhan) [inline sdk/attestation/azure-security-attestation/inc/azure/attestation/attestation_client.hpp:139]: I believe this is generally considered a non-breaking change. C++ (14 and lower) doesn't have named parameters, correct?

File: sdk/attestation/azure-security-attestation/inc/azure/attestation/attestation_client_options.hpp [hunk 6]
```diff
@@ -138,16 +138,16 @@ namespace Azure { namespace Security { namespace Attestation {
...
   class AttestationDataType final
       : public Azure::Core::_internal::ExtendableEnumeration<AttestationDataType> {
 
   public:
     /**
-     * @brief Construct a new attestation DataType object
+     * @brief Construct a new AttestationDataType object
      *
-     * @param dataType The string version for the Key Vault keys service.
+     * @param dataType The string version for the DataType object.
```
Comment (ahsonkhan) [inline sdk/attestation/azure-security-attestation/inc/azure/attestation/attestation_client_options.hpp:150]: Nice!

File: sdk/attestation/azure-security-attestation/src/attestation_administration_client.cpp [hunk 7]
```diff
@@ -534,7 +534,6 @@ AttestationAdministrationClient::RemoveIsolatedModeCertificate(
  * @details Validating the response returned by the attestation service requires a set of
  * possible signers for the attestation token.
  *
```
Comment (ahsonkhan) [inline sdk/attestation/azure-security-attestation/src/attestation_administration_client.cpp:536]: I don't think we need this comment block at all in `.cpp` files. It is redundant and could create confusion. We only need such documentation in the corresponding `.hpp` file, which we already have.

File: sdk/attestation/azure-security-attestation/src/attestation_client.cpp [hunk 8]
```diff
@@ -285,9 +285,6 @@ void AttestationClient::RetrieveResponseValidationCollateral(Azure::Core::Contex
 
 /** @brief Construct a new Attestation Client object
  *
- * @param endpoint The URL address where the client will send the requests to.
- * @param credential The authentication method to use (required for TPM attestation).
- * @param options The options to customize the client behavior.
  */
```
Comment (ahsonkhan) [inline sdk/attestation/azure-security-attestation/src/attestation_client.cpp:288]: Same here/elsewhere. Remove duplicate comments in source file. We don't want to maintain 2 places for the same thing that might start differing subtly.

File: sdk/core/azure-core-amqp/inc/azure/core/amqp/models/amqp_value.hpp [hunk 9]
```diff
@@ -91,6 +104,13 @@ namespace Azure { namespace Core { namespace Amqp { namespace Models {
   class AmqpComposite;
   class AmqpDescribed;
 
+  /** An AMQP value.
+   * @details An AMQP value is a polymorphic type that can be used to represent any AMQP type.
+   *
+   *
```
Comment (ahsonkhan) [inline sdk/core/azure-core-amqp/inc/azure/core/amqp/models/amqp_value.hpp:110]:
```suggestion
```

File: sdk/core/azure-core-amqp/src/amqp/connection_string_credential.cpp [hunk 10]
```diff
@@ -131,12 +131,6 @@ namespace Azure { namespace Core { namespace Amqp { namespace _internal {
   /**
    * @brief Gets an authentication token.
    *
-   * @param tokenRequestContext A context to get the token in.
-   * @param context A context to control the request lifetime.
-   *
-   * @return Authentication token.
-   *
-   * @throw Credentials::AuthenticationException Authentication error occurred.
    */
```
Comment (ahsonkhan) [inline sdk/core/azure-core-amqp/src/amqp/connection_string_credential.cpp:134]: Same here, remove this fully.

File: sdk/core/azure-core/inc/azure/core/http/win_http_transport.hpp [hunk 11]
```diff
@@ -171,17 +171,13 @@ namespace Azure { namespace Core {
...
        */
       WinHttpTransport(Azure::Core::Http::Policies::TransportOptions const& options);
 
       /**
        * @brief Implements the HTTP transport interface to send an HTTP Request and produce an
        * HTTP RawResponse.
        *
-       * @param context A context to control the request lifetime.
-       * @param request an HTTP request to be send.
-       * @return A unique pointer to an HTTP RawResponse.
```
Comment (ahsonkhan) [inline sdk/core/azure-core/inc/azure/core/http/win_http_transport.hpp:184]: Why do we have to remove these? Is it because of the function being virtual?

File: sdk/core/azure-core/src/http/user_agent.cpp [hunk 12]
```diff
@@ -2,9 +2,9 @@
 // SPDX-License-Identifier: MIT
 
 /**
- * @file
- * @brief HTTP pipeline is a stack of HTTP policies.
- * @remark See #policy.hpp
+ * @file user_agent.cpp
+ * @brief Contains the user agent string generator implementation.
+ * @remark See policy.hpp
```
Comment (ahsonkhan) [inline sdk/core/azure-core/src/http/user_agent.cpp:7]: Remove?
```suggestion
```

File: sdk/keyvault/azure-security-keyvault-certificates/inc/azure/keyvault/certificates/certificate_client_models.hpp [hunk 13]
```diff
@@ -1221,29 +1226,35 @@ namespace Azure { namespace Security { namespace KeyVault { namespace Certificat
...
   struct GetPropertiesOfCertificateVersionsOptions final
   {
+    /** @brief The token for the next page, if present. */
     Azure::Nullable<std::string> NextPageToken;
   };
 
   /**
-   * @brief The options for calling an operation #GetPropertiesOfIssuers
+   * @brief The options for calling an operation
+   * #Azure::Security::KeyVault::Certificates::CertificateClient::GetPropertiesOfIssuers
```
Comment (ahsonkhan) [inline sdk/keyvault/azure-security-keyvault-certificates/inc/azure/keyvault/certificates/certificate_client_models.hpp:1241]:
```suggestion
   * @brief The options for calling an operation
   * #Azure::Security::KeyVault::Certificates::CertificateClient::GetPropertiesOfIssuers.
```

File: sdk/keyvault/azure-security-keyvault-certificates/inc/azure/keyvault/certificates/certificate_client_models.hpp [hunk 14]
```diff
@@ -1221,29 +1226,35 @@ namespace Azure { namespace Security { namespace KeyVault { namespace Certificat
...
   struct GetPropertiesOfIssuersOptions final
   {
+    /** @brief The token for the next page, if present. */
     Azure::Nullable<std::string> NextPageToken;
   };
 
   /**
-   * @brief The options for calling an operation #GetDeletedCertificates
+   * @brief The options for calling an operation
+   * #Azure::Security::KeyVault::Certificates::CertificateClient::GetDeletedCertificates
```
Comment (ahsonkhan) [inline sdk/keyvault/azure-security-keyvault-certificates/inc/azure/keyvault/certificates/certificate_client_models.hpp:1252]:
```suggestion
   * @brief The options for calling an operation
   * #Azure::Security::KeyVault::Certificates::CertificateClient::GetDeletedCertificates.
```

File: sdk/keyvault/azure-security-keyvault-certificates/inc/azure/keyvault/certificates/certificate_client_models.hpp [hunk 15]
```diff
@@ -1168,6 +1168,10 @@ namespace Azure { namespace Security { namespace KeyVault { namespace Certificat
     ~CertificateOperationProperties() = default;
   };
 
+  /**
+   * @brief Represents a deleted certificate in a KeyVault
```
Comment (ahsonkhan) [inline sdk/keyvault/azure-security-keyvault-certificates/inc/azure/keyvault/certificates/certificate_client_models.hpp:1172]: Consistently end with a period.
```suggestion
   * @brief Represents a deleted certificate in a KeyVault.
```

File: sdk/keyvault/azure-security-keyvault-certificates/inc/azure/keyvault/certificates/certificate_client_operations.hpp [hunk 16]
```diff
@@ -79,7 +79,7 @@ namespace Azure { namespace Security { namespace KeyVault { namespace Certificat
      * @param resumeToken A previously generated token used to resume the polling of the
      * operation.
      * @param client A #CertificateClient that is used for getting status updates.
-     * @param context A #Azure::Core::Context controlling the request lifetime.
+     * @param context A Azure::Core::Context controlling the request lifetime.
```
Comment (ahsonkhan) [inline sdk/keyvault/azure-security-keyvault-certificates/inc/azure/keyvault/certificates/certificate_client_operations.hpp:82]: Should we be using **A** `Azure::Core::Context` or **An**?

File: sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keys/cryptography/cryptography_client.hpp [hunk 17]
```diff
@@ -42,7 +42,6 @@ namespace Azure {
    *
    */
   class CryptographyClient final {
-  protected:
```
Comment (ahsonkhan) [inline sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keys/cryptography/cryptography_client.hpp:45]: I don't think we can do this (breaking change).

**Review by ahsonkhan — COMMENTED** [review]
Reviewed 68/101 files. Thanks for fixing these!

File: sdk/storage/azure-storage-queues/inc/azure/storage/queues/queue_sas_builder.hpp [hunk 18]
```diff
@@ -48,12 +48,14 @@ namespace Azure { namespace Storage { namespace Sas {
     All = ~0,
   };
 
+  /** @brief Logical OR of two values*/
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-queues/inc/azure/storage/queues/queue_sas_builder.hpp:51]: This is actually bitwise OR (`|` in C++), not logical OR (`||` in C++)

File: sdk/storage/azure-storage-queues/inc/azure/storage/queues/queue_sas_builder.hpp [hunk 19]
```diff
@@ -48,12 +48,14 @@ namespace Azure { namespace Storage { namespace Sas {
...
   inline QueueSasPermissions operator|(QueueSasPermissions lhs, QueueSasPermissions rhs)
   {
     using type = std::underlying_type_t<QueueSasPermissions>;
     return static_cast<QueueSasPermissions>(static_cast<type>(lhs) | static_cast<type>(rhs));
   }
 
+  /** @brief Logical AND of two values*/
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-queues/inc/azure/storage/queues/queue_sas_builder.hpp:58]: ditto

**Review by Jinming-Hu — APPROVED** [review]
Changes to storage SDK LGTM.

My only concern is that all changes in auto-generated file `rest_client.hpp` will be discarded next time we update protocol layer.
Some of the changes were caused by missing docs in swagger, which we could fix.
The others, however, require code change in codeGen to fix.
image

**Comment by ahsonkhan** [comment]
> My only concern is that all changes in auto-generated file `rest_client.hpp` will be discarded next time we update protocol layer.
Some of the changes were caused by missing docs in swagger, which we could fix.

Could we update these comments into the swagger readme transform for storage, so they are retained on re-gen, while we wait for the swagger itself to be fixed and `description` fields added?

**Comment by Jinming-Hu** [comment]
> Could we update these comments into the swagger readme transform for storage, so they are retained on re-gen, while we wait for the swagger itself to be fixed and `description` fields added?

Sure, we can. This is less of a problem and I'm not worried about this.

But docs for enumerations requires updating codeGen. Do you know who maintains codeGen now? or it's not maintained anymore?

File: sdk/storage/azure-storage-queues/inc/azure/storage/queues/queue_sas_builder.hpp (hunk 18 above)
Comment (LarryOsterman) [reply sdk/storage/azure-storage-queues/inc/azure/storage/queues/queue_sas_builder.hpp:51]: Please propose a suggestion and I'll apply it.

**Comment by LarryOsterman** [comment]
> Changes to storage SDK LGTM.
> 
> My only concern is that all changes in auto-generated file `rest_client.hpp` will be discarded next time we update protocol layer. Some of the changes were caused by missing docs in swagger, which we could fix. The others, however, require code change in codeGen to fix. image

I was worried about this as I was making the changes. But IMHO having up-to-date documentation for everything in the SDK is a bigger deal - there were a LOT of fields that had absolutely no documentation, and while some of them were relatively obvious, others were not.

File: cmake-modules/AzureDoxygen.cmake (hunk 3 above)
Comment (LarryOsterman) [reply cmake-modules/AzureDoxygen.cmake:23]: I have an outstanding request to the ES team to upgrade to a current version of doxygen. Until then, we need to disable the failures :(.

File: cmake-modules/AzureDoxygen.cmake (hunk 4 above)
Comment (LarryOsterman) [reply cmake-modules/AzureDoxygen.cmake:33]: Because they contain macros that start with AZURE_<PACKAGE>_, and I don't think we want them in our product documentation. And I don't think we want to have to document the macros that are used for package building.

File: sdk/attestation/azure-security-attestation/inc/azure/attestation/attestation_client.hpp (hunk 5 above)
Comment (LarryOsterman) [reply sdk/attestation/azure-security-attestation/inc/azure/attestation/attestation_client.hpp:139]: That is correct - the C++ language doesn't have named parameters, unlike C#.

File: sdk/core/azure-core/inc/azure/core/http/win_http_transport.hpp (hunk 11 above)
Comment (LarryOsterman) [reply sdk/core/azure-core/inc/azure/core/http/win_http_transport.hpp:184]: Doxygen gets annoyed when the documentation for a function appears in multiple locations. This applies when the function is declared, it also applies when a virtual function overload is defined.

Comment (LarryOsterman) [reply sdk/core/azure-core/inc/azure/core/http/win_http_transport.hpp:184]: And ultimately, we had to remove them because doxygen complained about their existance.

File: sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keys/cryptography/cryptography_client.hpp (hunk 17 above)
Comment (LarryOsterman) [reply sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keys/cryptography/cryptography_client.hpp:45]: It's not a breaking change - the CryptographyClient is declared as final, so there can never be any classes derived from CryptographyClient.

**Comment by LarryOsterman** [comment]
> > Could we update these comments into the swagger readme transform for storage, so they are retained on re-gen, while we wait for the swagger itself to be fixed and `description` fields added?
> 
> Sure, we can. This is less of a problem and I'm not worried about this.
> 
> But docs for enumerations requires updating codeGen. Do you know who maintains codeGen now? or it's not maintained anymore?

Btw, doxygen is configured to ignore types in the _internal and _detail namespace, would it be possible to leverage that in some way in the codegen (putting the generated types in a _detail namespace and creating human written wrappers for the types we want to expose?)

**Review by antkmsft — APPROVED** [review]
Thank you!
We're not in as bad shape as I thought, although it was still a lot of work. I'm glad that from now on, we'll have the docs up-to-date, and we won't notice the cost of maintaining it, because it will be giving the errors right away, and there can't be that many at once.

File: sdk/core/azure-core-amqp/inc/azure/core/amqp/models/amqp_properties.hpp [hunk 20]
```diff
@@ -12,6 +12,14 @@
...
 
 struct PROPERTIES_INSTANCE_TAG;
+#if defined(__GNUC__) && (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6))
+#pragma GCC diagnostic push
+#elif defined(__clang__) // !__clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdocumentation-unknown-command"
+#endif // _MSC_VER
+
+///  @cond HIDDEN
```
Comment (antkmsft) [inline sdk/core/azure-core-amqp/inc/azure/core/amqp/models/amqp_properties.hpp:22]: The template that gets specialized is from `_internal` namespace, yet Doxygen still loses its mind?

Comment (LarryOsterman) [reply sdk/core/azure-core-amqp/inc/azure/core/amqp/models/amqp_properties.hpp:22]: doxygen doesn't get upset, clang gets upset. So I need to suppress the clang warning.

File: sdk/core/azure-core-amqp/src/models/amqp_error.cpp [hunk 21]
```diff
@@ -12,11 +12,25 @@
...
 #include <iostream>
 
+#if defined(__GNUC__) && (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6))
+#pragma GCC diagnostic push
+#elif defined(__clang__) // !__clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdocumentation-unknown-command"
+#endif // _MSC_VER
+
+/// @cond HIDDEN
```
[the remark this answers is not in the document]
Comment (LarryOsterman) [reply sdk/core/azure-core-amqp/src/models/amqp_error.cpp:22]: These aren't doxygen warnings. They're clang warnings.

And doxygen looks at all files - headers and implementations. That's the way it's worked since it was first brought online, and I'm loathe to change that significant a behavior.

There is something about the template specialization that doxygen is upset about, I believe it's because the type specialization is in the global namespace, but I'm not sure. So I work around it by asking doxygen to skip over the types.

And the failure is happening on the doxygen parse - doxygen compiles the C++ code and then uses the syntax tree generated to find all the documentation. That's *before* it gets to filtering out types.

File: sdk/core/azure-core/inc/azure/core/etag.hpp [hunk 22]
```diff
@@ -67,12 +67,12 @@ class ETag final {
...
   */
 
-  /*
-   * @brief Indicates whether two #Azure::Core::ETag values are equal.
-   * @param left #Azure::Core::ETag to compare.
-   * @param right #Azure::Core::ETag to compare.
-   * @param comparisonKind Determines what #Azure::Core::ETag::ETagComparison to perform, default
-   * is #Azure::Core::ETag::ETagComparison Strong.
+  /** @brief Indicates whether two #Azure::ETag values are equal.
+   *
```
Comment (RickWinter) [inline sdk/core/azure-core/inc/azure/core/etag.hpp:71]: Extra whitespace

File: sdk/keyvault/azure-security-keyvault-certificates/inc/azure/keyvault/certificates/certificate_client_operations.hpp (hunk 16 above)
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-certificates/inc/azure/keyvault/certificates/certificate_client_operations.hpp:82]: Missing anchor tag `#`
```suggestion
     * @param context A #Azure::Core::Context controlling the request lifetime.
```

File: sdk/keyvault/azure-security-keyvault-certificates/inc/azure/keyvault/certificates/certificate_client_operations.hpp [hunk 23]
```diff
@@ -90,14 +90,14 @@ namespace Azure { namespace Security { namespace KeyVault { namespace Certificat
     /**
      * @brief Cancels the operation.
      *
-     * @param context A #Azure::Core::Context controlling the request lifetime.
+     * @param context A Azure::Core::Context controlling the request lifetime.
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-certificates/inc/azure/keyvault/certificates/certificate_client_operations.hpp:93]: Why are you removing the anchor tag?

File: sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keys/cryptography/cryptography_client.hpp [hunk 24]
```diff
@@ -85,12 +84,12 @@ namespace Azure {
...
      *
-     * @param keyId The key identifier of the #KeyVaultKey which will be used for cryptographic
-     * operations.
-     * @param credential A #TokenCredential used to authenticate requests to the vault, like
-     * DefaultAzureCredential.
-     * @param options #CryptographyClientOptions for the #CryptographyClient for local or remote
-     * operations on Key Vault.
+     * @param keyId The key identifier of the #Azure::Security::KeyVault::Keys::KeyVaultKey which
+     * will be used for cryptographic operations.
+     * @param credential An Azure::Core::Credentials::TokenCredential used to authenticate requests
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keys/cryptography/cryptography_client.hpp:89]: Should this have an anchor tag?

**Review by RickWinter — APPROVED** [review]
Noticed in a lot of places you removed the anchor tag on the Azure::Core::Context.  Why?

**Comment by LarryOsterman** [comment]
> Noticed in a lot of places you removed the anchor tag on the Azure::Core::Context. Why?

Because doxygen can't resolve anchor tags in other packages without an explicit link. Annoying, but I can understand it.

File: sdk/keyvault/azure-security-keyvault-certificates/inc/azure/keyvault/certificates/certificate_client_operations.hpp (hunk 23 above)
Comment (LarryOsterman) [reply sdk/keyvault/azure-security-keyvault-certificates/inc/azure/keyvault/certificates/certificate_client_operations.hpp:93]: Because Azure::Core::Context is in another package, and doxygen cannot figure out the URL for a type defined in another package. This can be replaced with a permalink to the docs website, but just using an anchor doesn't work :(

File: sdk/core/azure-core-amqp/src/amqp/message_receiver.cpp [hunk 25]
```diff
@@ -27,12 +27,26 @@
 
 using namespace Azure::Core::Diagnostics::_internal;
 using namespace Azure::Core::Diagnostics;
+#if defined(__GNUC__) && (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6))
+#pragma GCC diagnostic push
+#elif defined(__clang__) // !__clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdocumentation-unknown-command"
+#endif // _MSC_VER
```
[the remark this answers is not in the document]
Comment (LarryOsterman) [reply sdk/core/azure-core-amqp/src/amqp/message_receiver.cpp:35]: IIRC there are very real challenges with preprocessor definitions within preprocessor macros, but I can try to see if I can get it to work.

File: sdk/keyvault/azure-security-keyvault-certificates/inc/azure/keyvault/certificates/certificate_client_operations.hpp (hunk 16 above)
Comment (LarryOsterman) [reply sdk/keyvault/azure-security-keyvault-certificates/inc/azure/keyvault/certificates/certificate_client_operations.hpp:82]: Anchor tags can't span packages :(.

File: sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keys/cryptography/cryptography_client.hpp (hunk 24 above)
Comment (LarryOsterman) [reply sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keys/cryptography/cryptography_client.hpp:89]: Yes, but anchor tags can't span packages :(.

File: sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/keyvault_secret.hpp [hunk 26]
```diff
@@ -10,6 +10,11 @@
 #include "azure/keyvault/secrets/keyvault_secret_properties.hpp"
 
 namespace Azure { namespace Security { namespace KeyVault { namespace Secrets {
+
+  /**
+   * @brief A Secret held in KeyVault.
+   *
+   */
   struct KeyVaultSecret
   {
```
Comment (ahsonkhan) [inline sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/keyvault_secret.hpp:19]: I believe this fixes https://github.com/Azure/azure-sdk-for-cpp/issues/4691

Consider borrowing the wording from Java, which reads pretty well, imo.

Here's the doc description from the rest-api-spec:
https://github.com/Azure/azure-rest-api-specs/blob/c7f3e601fd326ca910c3d2939b516e15581e7e41/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.4/keys.json#L1510

Here's an example of wording from .NET:
https://github.com/Azure/azure-sdk-for-net/blob/1d49335c35d20bc812107a6e67187cab58b9d58f/sdk/keyvault/Azure.Security.KeyVault.Secrets/src/KeyVaultSecret.cs#L10

From Java:
https://learn.microsoft.com/en-us/java/api/com.azure.security.keyvault.secrets.models.keyvaultsecret?view=azure-java-stable
> Secret is the resource consisting of name, value and its attributes specified in [SecretProperties](https://learn.microsoft.com/en-us/java/api/com.azure.security.keyvault.secrets.models.secretproperties?view=azure-java-stable). It is managed by Secret Service.

File: sdk/core/azure-core-amqp/src/amqp/message_receiver.cpp (hunk 25 above)
[the remark this answers is not in the document]
Comment (LarryOsterman) [reply sdk/core/azure-core-amqp/src/amqp/message_receiver.cpp]: FIgured out how to make it work.

**Comment by LarryOsterman** [comment]
Fixes #4691

File: sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/keyvault_secret.hpp [hunk 27]
```diff
@@ -10,6 +10,12 @@
 #include "azure/keyvault/secrets/keyvault_secret_properties.hpp"
 
 namespace Azure { namespace Security { namespace KeyVault { namespace Secrets {
+
+  /**
+   * @brief Secret is the resource consisting of name, value and its attributes specified in
+   * SecretProperties. It is managed by Secret Service.
+   *
+   */
```
Comment (ahsonkhan) [inline sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/keyvault_secret.hpp:18]: This fixes https://github.com/Azure/azure-sdk-for-cpp/issues/4691
